contribs/osm/src/main/java/org/matsim/contrib/osm/networkReader/SupersonicOsmNetworkReader.java

```diff
@@ -299,6 +300,14 @@ private double getFreespeed(Map<String, String> tags, double linkLength, LinkPro
         }
     }
 
+    private double getMaxSpeed(Map<String, String> tags, double linkLength, LinkProperties properties) {
+        if (tags.containsKey(OsmTags.MAXSPEED)) {
+            return parseSpeedTag(tags.get(OsmTags.MAXSPEED), properties);
+        } else {
+            return LinkProperties.calculateSpeedIfNoSpeedTag(linkLength, properties);
```

Comment:
here I use the same logic as in getFreespeed. Alternatively, if ther is no speed tag available, one could also ommit to set the attribute, since it is always going to be the same as the freespeed.